Upgrade emotion so we can upgrade emotion-icons to get latest icons.

vdonato opened the pull request:
## 📚 Context

This PR was originally a commit in #4504, but we might as well merge it straight
into `develop` instead of `feature/multipage-apps` as there's no reason we can't
upgrade these packages now.

- What kind of change does this PR introduce?

  - [x] Other, please describe: package upgrades

## 🧠 Description of Changes

- Upgrade a bunch of emotion-related packages
- Tweak imports to account for changes in emotion major versions

tvst committed:
Upgrade emotion so we can upgrade emotion-icons to get latest icons.

vdonato committed:
Update snapshots and fix tests

lukasmasuch reviewed (approved):
LGTM 👍 There is just one e2e test failing :) Just a curious question, how do we trigger the update of the `NOTICES` file?

tconkling reviewed (approved):
Looks good on the dependencies front - this just updates to later versions of dependencies we already use

vdonato committed:
Switch back to the old hamburger menu icon for now

We'll change to the material icon in the multipage apps feature branch along
with the sidebar icons. This should keep everything in this PR visually
identical so that it's a true package version upgrade PR.

vdonato commented:
> LGTM 👍 There is just one e2e test failing :) Just a curious question, how do we trigger the update of the NOTICES file?

I'm actually not sure since this PR was made by cherry-picking a commit out of a PR that @tvst made, so I didn't update the NOTICES file myself 😅
